app/views/ile_query_ducr_response.scala.html

```diff
@@ -14,73 +14,87 @@
+)
 
-@(info: DucrInfo)(implicit request: Request[_], messages: Messages)
+@(info: DucrInfo, parentMucr: Option[MucrInfo])(implicit request: Request[_], messages: Messages)
 
+@govukLayout(
+    useCustomContentWidth = true,
+    title = Title("ileQueryResponse.ducr.title"),
+    backButton = Some(BackButton(messages("site.back"), controllers.ileQuery.routes.IleQueryController.displayQueryForm()))) {
 
+    <div class="govuk-grid-row">
 
-@main_template(title = Title("ileQueryResponse.ducr.title")) {
+        <div class="govuk-grid-column-two-thirds">
+            <h1 class="govuk-heading-xl govuk-!-margin-bottom-2">@messages("ileQueryResponse.ducr.title")</h1>
+            <span class="govuk-caption-xl govuk-!-margin-bottom-9">@info.ucr</span>
+        </div>
+        <div class="govuk-grid-column-two-thirds">
+            <h2 class="govuk-heading-m">@messages("ileQueryResponse.details")</h2>
 
-  @components.back_link(controllers.ileQuery.routes.IleQueryController.displayQueryForm())
+            @govukSummaryList(SummaryList(
+                attributes = Map("id" -> "summary"),
+                rows = Seq(
+                    SummaryListRow(
+                        key = Key(
+                            content = Text(messages("ileQueryResponse.details.route"))
+                        ),
+                        value = Value(
+                            content = info.entryStatus.flatMap(_.roe).map(ileCodeMapper.routeOfEntry(_)).getOrElse(Empty)
+                        )
+                    ),
+                    SummaryListRow(
+                        key = Key(
+                            content = Text(messages("ileQueryResponse.details.entryStatus"))
+                        ),
+                        value = Value(
+                            content = info.entryStatus.flatMap(_.soe).map(ileCodeMapper.statusOfEntryDucr(_)).getOrElse(Empty)
+                        )
+                    ),
+                    SummaryListRow(
+                        key = Key(
+                            content = Text(messages("ileQueryResponse.details.transport"))
+                        ),
+                        value = Value(
+                            content = info.transport.map(transportMapper.transportHtml).getOrElse(Empty)
+                        )
+                    ),
+                    SummaryListRow(
+                        key = Key(
+                            content = Text(messages("ileQueryResponse.details.inputCustomsStatus"))
+                        ),
+                        value = Value(
+                            content = info.entryStatus.flatMap(_.ics).map(ileCodeMapper.inputCustomsStatus(_)).getOrElse(Empty)
+                        )
+                    )
+                ).filterNot(row => row.key.content == Text(messages("ileQueryResponse.details.transport")) && row.value.content == Empty),
+                classes = "govuk-!-margin-bottom-9"
+            ))
+        </div>
```

Comment:
Can this part be also extracted into a component? Not to have multiple levels of abstraction here.